CMakeLists.txt

```diff
@@ -140,30 +157,22 @@ if( DETRAY_SETUP_BENCHMARK )
    endif()
 endif()
 
-cmake_dependent_option(DETRAY_BENCHMARKS_MULTITHREAD "Enable multithreaded benchmarks" Off "DETRAY_BENCHMARKS" Off)
-if(DETRAY_BENCHMARKS_MULTITHREAD)
-    message(STATUS "Using multithreaded benchmarks")
-endif()
-if(DETRAY_BENCHMARKS AND DETRAY_BENCHMARKS_REP)
-    message(STATUS "Repetitions for benchmarks: " ${DETRAY_BENCHMARKS_REP})
+# Set up all of the libraries of the project.
+add_subdirectory( core )
+add_subdirectory( plugins )
+add_subdirectory( io )
+if( DETRAY_DISPLAY )
+   add_subdirectory( display )
 endif()
 
-# Set GPU-related options
-if(DETRAY_BUILD_CUDA)
-  enable_language(CUDA)
-endif()
-
-add_subdirectory(apps)
-add_subdirectory(core)
-add_subdirectory(plugins)
-add_subdirectory(io)
+# Set up the project's applications.
+add_subdirectory( apps )
```

Comment:
I think it is mostly the display code that lies in the apps directory, so this kind of belongs to the ```DETRAY_DISPLAY``` option in my mind. I am not sure why this is split up currently

Reply:
I didn't want to re-organise this part of the repository as part of this PR. But we should absolutely do that in a follow-up PR.